tests/check_vcs_permalinks_test.py

```diff
@@ -20,17 +22,27 @@ def test_passing(tmpdir):
     assert not main((f.strpath,))
 
 
-def test_failing(tmpdir, capsys):
+@pytest.mark.parametrize(
+    'contents',
+    (
+        'http://github.com/asottile/test/blob/master/foo#L1',
+        'http://www.github.com/asottile/test/blob/master/foo#L1',
+        'https://github.com/asottile/test/blob/master/foo#L1',
+        'https://www.github.com/asottile/test/blob/master/foo#L1',
+        'https://www.github.com/asottile/test/blob/master/foo#my-anchor',
```

Comment:
This arguably _shouldn't_ be a permalink.  For example https://github.com/asottile/pyupgrade/blob/master/README.md#dictionary-comprehensions

Reply:
Can you clarify what you mean? https://github.com/asottile/pyupgrade/blob/a8bdf45b6b91b8465cc739d350d56c7fb260a8db/README.md#dictionary-comprehensions is a valid link

Reply:
(I hit submit on the comment too fast) I was thinking that someone could remove the anchor in a future commit

Reply:
The hook is intended to catch links to code, if documentation breaks headers that's probably a problem with the documentation (I would, for instance, link to [latest:parametrize](https://docs.pytest.org/en/latest/example/parametrize.html#different-options-for-test-ids) preferably over [2.7.3:parametrize](https://docs.pytest.org/en/2.7.3/example/parametrize.html#different-options-for-test-ids) (old docs))

The point here being that linking to a specific line of code against a master branch is _always_ wrong, but linking to a header on some documentation is most of the time just fine.

The general design of hooks I maintain aim to *minimize* false-positives

Reply:
Gotcha, thanks for the context